File: pair_impair.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

// Demande à l'utilisateur de saisir un nombre
rl.question('Veuillez saisir un nombre: ', (nombre) => {
  // Calcule le reste de la division par 2
  const reste = nombre % 2;

  // Vérifie si le reste est égal à 0
  if (reste === 0) {
    console.log("ce nombre pair");
  } else {
    console.log("ce nombre impair");
  }

  // Ferme l'interface readline
  rl.close();
});

File: pair_impair.java
import java.util.Scanner;

public class pair_impair {

	public static void main(String[] args) {
		
		// déclaration de variable
		double nombre, reste;
		
		//enregistremenet des variables tapées par l'utilisateur
		Scanner clavier = new Scanner(System.in);
		System.out.print("Veuillez rentrer un nombre: ");
		nombre = clavier.nextDouble();
		reste = nombre%2;
		
		//La condition
		if(reste==0) {
			System.out.print("le nombre "+nombre+" est pair");
		}else {
			System.out.print("le nombre "+nombre+" est impair");
		}
		
		clavier.close();

	}
}
